### ISSUE TITLE
HA does not find sonoff

### ISSUE THREAD

**Issue by frayberen** [AlexxIT/SonoffLAN#573]
Hi Alex. Excuse my English, I'm not strong at it. I've been trying for three days to make it work, but there's no way to do it. I put below the debug  log
Any idea what the problem may be? Thanks in advance.

```
DEBUG    main          SysInfo: {'version': '2021.8.6', 'dev': False, 'hassio': True, 'virtualenv': False, 'python_version': '3.9.6', 'docker': True, 'user': 'root', 'arch': 'x86_64', 'os_name': 'Linux', 'os_version': '4.19.0-17-amd64', 'supervisor': '2021.06.8', 'host_os': 'Debian GNU/Linux 10 (buster)', 'docker_version': '20.10.8', 'chassis': 'desktop'}
2021-08-15 00:20:50  DEBUG    main          CLOUD mode start
2021-08-15 00:20:59  DEBUG    sonoff_cloud  6 devices loaded from the Cloud Server
2021-08-15 00:20:59  DEBUG    main          10004cfbff == Init   | {'uiid': 14, 'extra': {'manufacturer': 'Sonoff', 'model': 'Basic', 'sw_version': 'ITA-GZ1-GL v3.5.0'}, 'params': {'init': 1, 'sledOnline': 'on', 'version': 8, 'bindInfos': {'gaction': ['...', '...', '...'], 'alexa': []}, 'pulseWidth': 30000, 'pulse': 'on', 'timers': [], 'rssi': -53, 'staMac': '...', 'startup': 'on', 'fwVersion': '3.5.0', 'switch': 'off', 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2021-08-15 00:20:59  DEBUG    main          10004a4219 == Init   | {'uiid': 14, 'extra': {'manufacturer': 'Sonoff', 'model': 'Basic', 'sw_version': 'ITA-GZ1-GL v3.5.0'}, 'params': {'pulseWidth': 500, 'pulse': 'off', 'init': 1, 'sledOnline': 'on', 'version': 8, 'bindInfos': {'gaction': ['...', '...', '...'], 'alexa': []}, 'rssi': -57, 'staMac': '...', 'startup': 'on', 'fwVersion': '3.5.0', 'switch': 'off', 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2021-08-15 00:20:59  DEBUG    main          10004cf52f == Init   | {'uiid': 14, 'extra': {'manufacturer': 'Sonoff', 'model': 'Basic', 'sw_version': 'ITA-GZ1-GL v3.5.0'}, 'params': {'pulseWidth': 500, 'pulse': 'off', 'init': 1, 'sledOnline': 'on', 'version': 8, 'bindInfos': {'gaction': ['...', '...', '...'], 'alexa': []}, 'rssi': -64, 'staMac': '...', 'startup': 'on', 'fwVersion': '3.5.0', 'switch': 'off', 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2021-08-15 00:20:59  DEBUG    main          10004cfb9c == Init   | {'uiid': 14, 'extra': {'manufacturer': 'Sonoff', 'model': 'Basic', 'sw_version': 'ITA-GZ1-GL v3.5.0'}, 'params': {'bindInfos': {'alexa': [], 'gaction': ['...']}, 'pulseWidth': 500, 'pulse': 'off', 'init': 1, 'sledOnline': 'on', 'version': 8, 'switch': 'off', 'fwVersion': '3.5.0', 'startup': 'off', 'staMac': '...', 'rssi': -62, 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2021-08-15 00:20:59  DEBUG    main          10004cfaa0 == Init   | {'uiid': 14, 'extra': {'manufacturer': 'Sonoff', 'model': 'Basic', 'sw_version': 'ITA-GZ1-GL v3.5.0'}, 'params': {'bindInfos': {'alexa': [], 'gaction': ['...']}, 'pulseWidth': 500, 'pulse': 'off', 'init': 1, 'sledOnline': 'on', 'version': 8, 'switch': 'off', 'fwVersion': '3.5.0', 'startup': 'off', 'staMac': '...', 'rssi': -41, 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2021-08-15 00:20:59  DEBUG    main          10007e71c9 == Init   | {'uiid': 1, 'extra': {'manufacturer': 'Sonoff', 'model': 'Basic2', 'sw_version': 'PSF-B01-GL v3.5.1'}, 'params': {'sledOnline': 'on', 'switch': 'off', 'fwVersion': '3.5.1', 'rssi': -40, 'staMac': '...', 'startup': 'off', 'init': 1, 'pulse': 'off', 'pulseWidth': 500, 'version': 8, 'bindInfos': {'gaction': ['...'], 'alexa': []}, 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2021-08-15 00:20:59  DEBUG    sonoff_cloud  Cloud init: {'error': 0, 'apikey': '...', 'config': {'hb': 1, 'hbInterval': 145}, 'sequence': '1628979659560'}
```

**Comment by AlexxIT** [comment]
As I can see, you have 6 devices and all of them should work

**Comment by frayberen** [comment]
They should, but they don't work. When I reboot HA it says that it cannot find the devices. The last line of the log shows a cloud init error. I have two installations at different sites. Everything is exactly the same. in one place it works and in another it does not. can it be a question of ports? Lately Vodafone is doing too many mischief with them. In this installation I have had Everything is exactly the same. in one place it works and in another it does not. can it be a question of ports? Lately Vodafone is doing too many mischief with them. In this installation I have had to enable 4443 by 443. They have closed it without saying anything. At the moment, I'm still looking ...

**Comment by AlexxIT** [comment]
Error 0 means no error. Maybe you have any errors in Hass logs? Maybe you have any Hass notification about Sonoff?

**Comment by frayberen** [comment]
No, I do not have any.

**Comment by dish67** [comment]
I had the same problem and my solution was that I stop the HA server and start the server. all work now :-)

**Comment by frayberen** [comment]
> Tuve el mismo problema y mi solución fue detener el servidor HA e iniciar el servidor. todo funciona ahora :-)

**Comment by frayberen** [comment]
I have rebooted at least 20 times and everything remains the same ... :((

**Comment by Tamsy** [comment]
Have you commented (disabled) "mode: local" insidde configuration.yaml?

This could be related to https://github.com/AlexxIT/SonoffLAN/issues/557

With HA core 2021.8.1 to core 2021.8.5 SonoffLAN only detected some Sonoff devices correctly when "mode: local" was disabled.

This seemed to have been fixed with HA core 2021.8.6 although Zeroconf threw (harmless) error messages to the logs.

With HA core 2021.8.7 those Zeroconf error messages went away (Zeroconf was bumped to v.0.35) but the latest SonoffLAN works only correctly if "mode: local" is disabled again. I havent tried whether activating  mode: local will work with Zeroconf v.0.35.1 or v.0.36.

**Comment by AlexxIT** [comment]
Probably fixed in latest master-version
